### DISCUSSION TITLE
GNS3 VM on VMware Fusion tech preview for M1

### DISCUSSION THREAD

**Discussion by grossmj** [GNS3/gns3-gui#3261] [Show and tell]
I built a GNS3 VM for ARM which can run on Apple silicon M1. You can download the disks (VMDK) [here](https://github.com/GNS3/gns3-gui/releases/download/v2.2.28/GNS3.VM.ARM64.2.2.28.zip)

**Please note this is EXPERIMENTAL**, do not expect it to work  like on Intel/AMD. VMware Fusion for M1 itself is just a preview. See below for limitations, workarounds and other issues.

### Create a custom virtual machine

It is not possible to import an OVA at the moment, see https://github.com/GNS3/gns3-gui/issues/3257#issuecomment-967789420

![Screenshot from 2021-11-18 19-01-48](https://user-images.githubusercontent.com/4752614/142552930-f49b20e0-568d-4394-a0c3-dd35044c7dcb.png)

### Choose the operating system

The GNS3 VM is based on Ubuntu 20.04 LTS for ARM64 architecture (aarch64).

![Screenshot from 2021-11-18 19-02-36](https://user-images.githubusercontent.com/4752614/142553094-a0a998e0-0d83-4c2b-b493-51165d27dffc.png)

### Choose a virtual disk

Select the first GNS3 VM disk.

![Screenshot from 2021-11-18 19-02-55](https://user-images.githubusercontent.com/4752614/142553414-64e354b4-82f2-4221-bef1-2752f1b584ca.png)

![Screenshot from 2021-11-18 21-00-56](https://user-images.githubusercontent.com/4752614/142553834-2e25276a-8ad2-4602-9e27-53c2a94ec2cd.png)

![Screenshot from 2021-11-18 21-01-19](https://user-images.githubusercontent.com/4752614/142554107-ef523274-8d98-4b80-b455-a0d1d37dbed6.png)

### Customize the name

![Screenshot from 2021-11-18 21-02-09](https://user-images.githubusercontent.com/4752614/142554214-6d526098-c7f4-431e-a1b7-c9caa7312d37.png)

### Add the second hard drive

Go to the VM settings and add an additional device

![Screenshot from 2021-11-18 21-04-09](https://user-images.githubusercontent.com/4752614/142554318-34e9a9fe-8eda-4125-b333-47819dfcfadb.png)

![Screenshot from 2021-11-18 21-04-33](https://user-images.githubusercontent.com/4752614/142554443-32283ff3-4b25-4602-9492-d544f4169729.png)

Select the second disk

![Screenshot from 2021-11-18 21-04-59](https://user-images.githubusercontent.com/4752614/142554473-7d31dbcd-3dcd-4256-86fc-1a1c266ce4ff.png)

![Screenshot from 2021-11-18 21-05-46](https://user-images.githubusercontent.com/4752614/142554737-33801b2e-8e0d-45dc-bbe8-5dd6c347cefc.png)

### Configure network adapters

Go back to the VM settings and add another device

![Screenshot from 2021-11-18 22-06-17](https://user-images.githubusercontent.com/4752614/142554885-29adff32-b145-415e-bb3d-a2b8385f1dcb.png)

Make sure it is configured for NAT

![Screenshot from 2021-11-18 22-08-13](https://user-images.githubusercontent.com/4752614/142554986-d29ae08a-70b6-4ae3-aa2e-af3c24985326.png)

Go to the first adapter settings and make sure it is configured to be private

![Screenshot from 2021-11-18 22-08-29](https://user-images.githubusercontent.com/4752614/142555072-405bf404-36bd-4aa3-afad-e68176ad24f3.png)

### Start the VM

![Screenshot from 2021-11-19 00-45-41](https://user-images.githubusercontent.com/4752614/142555198-de3402f9-26ee-41b2-96ad-7f6672615672.png)

![Screenshot from 2021-11-19 13-12-40](https://user-images.githubusercontent.com/4752614/142556113-1de1871b-5f1c-499c-998b-ee10c4b0d0c9.png)

### Launch GNS3 and use the VM as a remote server

![Screenshot from 2021-11-19 15-19-35](https://user-images.githubusercontent.com/4752614/142566792-27e817da-5945-4f77-9e8c-b1193526cecc.png)

# Limitations and workarounds

### Docker

Trying to run a container out of the box will result in an exec format error because the busybox binary that is distributed part of the GNS3 server package is compiled for the x86_64 architecture, not ARM.

![Screenshot from 2021-11-19 13-53-12](https://user-images.githubusercontent.com/4752614/142563413-6febd467-8bb6-4e44-8920-c75373a5fdfe.png)

![Screenshot from 2021-11-19 13-53-35](https://user-images.githubusercontent.com/4752614/142563431-c843bcf6-a7fd-452a-be34-c499a473df5e.png)

The workaround is to SSH to the GNS3 VM and type this command:

```sudo cp /usr/bin/busybox /usr/local/lib/python3.8/dist-packages/gns3server/compute/docker/resources/bin/busybox```

Note that this will not be needed with GNS3 version >= 3.0, the server will automatically use the host busybox: https://github.com/GNS3/gns3-server/pull/1908

However, any image or container you want to run must be built for the arm64v8 architecture, most images like Ubuntu, Alpine etc. support this. You can quickly check on Docker Hub what architectures are supported, for instance https://hub.docker.com/_/ubuntu

Compare this with a container we distribute, https://hub.docker.com/r/gns3/network_automation/tags, which only supports amd64. We aim to have our containers built for multiple architectures at a later stage.

This is what happens if you want to run Ubuntu image made for amd64 architecture:

```
gns3@gns3vm:~$ docker run --rm -t amd64/ubuntu uname -m
Unable to find image 'amd64/ubuntu:latest' locally
latest: Pulling from amd64/ubuntu
7b1a6ab2e44d: Pull complete 
Digest: sha256:7cc0576c7c0ec2384de5cbf245f41567e922aab1b075f3e8ad565f508032df17
Status: Downloaded newer image for amd64/ubuntu:latest
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
standard_init_linux.go:228: exec user process caused: exec format error
```

It is currently only possible to run images / containers made for other architectures on a x86_64 host, see https://www.stereolabs.com/docs/docker/building-arm-container-on-x86/ and https://github.com/multiarch/qemu-user-static#supported-host-architectures.

**Conclusion** it is only possible to Docker containers made for arm64/v8

### Qemu

Qemu is the big issue because KVM hardware acceleration is not supported.

My understanding is that to use KVM, the firmware must start Linux in hypervisor (EL2 / HYP) mode and the kernel should print something like this:

```
CPU: All CPU(s) started at EL2
...
kvm [1]: Hyp mode initialized successfully
```

As you can see GNS3 VM doesn't start with Hypervisor mode or EL2:

![Screenshot from 2021-11-19 00-51-05](https://user-images.githubusercontent.com/4752614/142556454-4c8a6b1a-988c-451a-be7f-43f3d7eed35d.png)

Also, I read the following `nested virtualization including the ARMv8.4 improvements are present on Apple M1. However, Apple's virtualization solution doesn't currently support those.`

Also, I was successful to run the GNS3 VM using a patched QEMU leveraging Hypervisor.framework (hvf acceleration): see https://gist.github.com/akihikodaki/87df4149e7ca87f18dc56807ec5a1bc5 or https://gist.github.com/citruz/9896cd6fb63288ac95f81716756cb9aa

However, as expected, nested virtualization isn't possible even after trying to enable it on the Qemu command line (virtualization=on).

![Screenshot from 2021-11-19 19-05-05](https://user-images.githubusercontent.com/4752614/142591463-d8f8b8f0-c1fd-4441-bc7d-9929bb3ab38d.png)

So I don't expect we will see nested virtualization until Apple or VMware does something.

I was able to manually create a template for IOSv-L2 and I had to force to not using KVM by adding `-machine accel=tcg` otherwise the server would complain that KVM is not available.

![Screenshot from 2021-11-19 19-34-11](https://user-images.githubusercontent.com/4752614/142595868-f9c528c3-b6f2-4b59-8a5e-f59dfb594e69.png)

![Screenshot from 2021-11-19 19-34-47](https://user-images.githubusercontent.com/4752614/142596050-18d0fae9-279c-4ca4-bf84-84429a1616bb.png)

**Conclusion**: supported but likely to be really slow

### Dynamips

Dynamips seems to work okay. I only tested with a c3725 IOS image.

![Screenshot from 2021-11-19 16-26-17](https://user-images.githubusercontent.com/4752614/142572701-73a5738d-ebf6-46a8-b441-56ae9067d79f.png)

However, Dynamips has been compiled without JIT (Just In Time) support which may slow things down or not be as optimized.

![Screenshot from 2021-11-19 13-13-56](https://user-images.githubusercontent.com/4752614/142569104-80369480-0bcc-44e4-85cf-3d09ba1fd869.png)

**Conclusion**: supported but may be slower or unstable

### IOU

It is not possible to run IOU because IOU images are compiled for i386 or x86_64. I remember of some Solaris Sparc images back in the days but nothing about ARM.

```
gns3@gns3vm:/opt/gns3/images/IOU$ file i86bi-linux-l3-adventerprisek9-ms.155-2.T.bin
i86bi-linux-l3-adventerprisek9-ms.155-2.T.bin: ELF 32-bit LSB executable, Intel 80386, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux.so.2, for GNU/Linux 2.6.9, stripped
```

```
gns3@gns3vm:/opt/gns3/images/IOU$ ./i86bi-linux-l3-adventerprisek9-ms.155-2.T.bin 
bash: ./i86bi-linux-l3-adventerprisek9-ms.155-2.T.bin: cannot execute binary file: Exec format error
```

One workaround is to use Qemu User Emulation: https://wiki.debian.org/QemuUserEmulation

I was successful to manually running an IOU image:

```
gns3@gns3vm:/opt/gns3/images/IOU$ qemu-i386-static i86bi-linux-l3-adventerprisek9-ms.155-2.T.bin 1
...

Cisco IOS Software, Linux Software (I86BI_LINUX-ADVENTERPRISEK9-M), Version 15.5(2)T, DEVELOPMENT TEST SOFTWARE
Technical Support: http://www.cisco.com/techsupport
Copyright (c) 1986-2015 by Cisco Systems, Inc.
Compiled Thu 26-Mar-15 07:36 by prod_rel_team

...

Linux Unix (Intel-x86) processor with 87588K bytes of memory.Installed image archive

Processor board ID 2048001
8 Ethernet interfaces
8 Serial interfaces
64K bytes of NVRAM.
```

and after installing `qemu-user-binfmt` I was able to execute IOU transparently.

```
gns3@gns3vm:/opt/gns3/images/IOU$ ./i86bi-linux-l3-adventerprisek9-ms.155-2.T.bin 1
...

Cisco IOS Software, Linux Software (I86BI_LINUX-ADVENTERPRISEK9-M), Version 15.5(2)T, DEVELOPMENT TEST SOFTWARE
Technical Support: http://www.cisco.com/techsupport
Copyright (c) 1986-2015 by Cisco Systems, Inc.
Compiled Thu 26-Mar-15 07:36 by prod_rel_team
*Nov 19 07:59:55.937: %CRYPTO-6-ISAKMP_ON_OFF: ISAKMP is OFF
*Nov 19 07:59:55.937: %CRYPTO-6-GDOI_ON_OFF: GDOI is OFF
Router>
```

I will write a howto set Qemu User Emulation up in the GNS3 VM.

**Conclusion**: supported but may be slow since it is emulation

### VPCS

No issue here since the vpcs Debian package was compiled for ARM.

```
gns3@gns3vm:~$ file /usr/bin/vpcs 
/usr/bin/vpcs: ELF 64-bit LSB shared object, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, BuildID[sha1]=2403ecb5c20e3f8264d0bafa15c022dd6e7b6759, for GNU/Linux 3.7.0, stripped
```

```
gns3@gns3vm:~$ vpcs 

Welcome to Virtual PC Simulator, version 0.8.2
Dedicated to Daling.
Build time: Aug 23 2021 11:15:00
Copyright (c) 2007-2015, Paul Meng (mirnshi@gmail.com)
All rights reserved.

VPCS is free software, distributed under the terms of the "BSD" licence.
Source code and license can be found at vpcs.sf.net.
For more information, please visit wiki.freecode.com.cn.

Press '?' to get help.

VPCS[1]>
```

**Conclusion**: supported, no issues detected

**Comment by jnatschev** [comment]
Wow! Nice.



Cheers,
John Natschev

**Reply by grossmj** [reply]
Haha no worries, hang on I am going to post about limitations / workarounds ;)

**Reply by grossmj** [reply]
I have updated the post

**Comment by grossmj** [comment]
Here are the instructions to run IOU

SSH to the VM, install Qemu User Emulation

```
sudo apt update
sudo apt install qemu binfmt-support qemu-user qemu-user-binfmt
```

Edit `/etc/apt/sources.list` and replace by this content:

```
deb [arch=i386] http://archive.ubuntu.com/ubuntu/ focal main restricted universe multiverse
deb [arch=i386] http://archive.ubuntu.com/ubuntu/ focal-updates main restricted universe multiverse
deb [arch=i386] http://archive.ubuntu.com/ubuntu/ focal-backports main restricted universe multiverse
deb [arch=i386] http://security.ubuntu.com/ubuntu/ focal-security main restricted universe multiverse

deb [arch=arm64] http://ports.ubuntu.com/ubuntu-ports/ focal main restricted universe multiverse
deb [arch=arm64] http://ports.ubuntu.com/ubuntu-ports/ focal-updates main restricted universe multiverse
deb [arch=arm64] http://ports.ubuntu.com/ubuntu-ports/ focal-backports main restricted universe multiverse
deb [arch=arm64] http://ports.ubuntu.com/ubuntu-ports/ focal-security main restricted universe multiverse
deb [arch=arm64] https://download.docker.com/linux/ubuntu focal stable
```

Finally, install gns3-iou which will in turn install the necessary IOU dependencies (SSL lib)

```
sudo dpkg --add-architecture i386
sudo apt update
sudo apt install libc6:i386 gns3-iou
```

![Screenshot from 2021-11-19 22-51-44](https://user-images.githubusercontent.com/4752614/142624099-182adc96-8563-487f-adf2-0e0970fe0da1.png)

**Reply by Ondjultomte** [reply]
Is this running ios locally ? even though perhaps slow ?

I need to run some sort of IOS locally on my m1 mac, just for fast testing when scripting. For full test I'll connect to my real lab setup.

**Comment by grossmj** [comment]
Regarding Qemu and the lack of KVM support in the VM, I think the problem is even bigger because even if we have it, this would only work for ARM (qemu-system-aarch64) not for qemu-system-x86_64, the major issue is that most virtual appliances from the big companies are all x86_64 based.

**Reply by ghost** [reply]
An ARM based computer like the M1 Macs will never be able to run x86 code in hardware, just because its CPU doesn't understand x86 code. So x86 code can only be executed by software emulation, what will be slow. As long as the major network vendors don't offer VM images for ARM based computer, buying a M1 Mac is not an option for me. Any decent intel computer will outperform a M1 Mac, when running x86 VMs.

**Reply by grossmj** [reply]
Maybe network vendors will switch to ARM or more realistically have both x86 and ARM appliances but this will most likely take years. So, indeed I wouldn't recommended buying a M1 Mac if you need to run x86 VMs.

**Comment by adilnas62711411** [comment]
Hello,

I downloaded the experimental vm and set it up in Vmware fusion tech preview per your instructions and have observed that the vm is not booting properly.

<img width="1022" alt="image" src="https://user-images.githubusercontent.com/29047168/142786852-ab2e1ab2-d9e5-46ab-a1a7-3f497fd86800.png">

**Reply by grossmj** [reply]
Please can you post a screenshot of your VM settings or even better, the content of the .vmx file inside the VM folder?

**Reply by adilnas62711411** [reply]
Here you go...

<img width="468" alt="image" src="https://user-images.githubusercontent.com/29047168/142946932-e4e6928c-3d98-47c6-a590-012dc5254baf.png">

<img width="468" alt="image" src="https://user-images.githubusercontent.com/29047168/142946945-a76333f2-c158-4e46-bc35-52b02052d040.png">


<img width="402" alt="image" src="https://user-images.githubusercontent.com/29047168/142946961-d2e5f62b-48c9-42c7-9dae-e1f4951bdf01.png">


<img width="468" alt="image" src="https://user-images.githubusercontent.com/29047168/142946976-06a5feb6-0f3a-488c-b697-a2be5f4abae8.png">


<img width="468" alt="image" src="https://user-images.githubusercontent.com/29047168/142946991-2c64a374-c49a-4b6d-8dc1-3c7c1980442d.png">


<img width="468" alt="image" src="https://user-images.githubusercontent.com/29047168/142947008-4d54692d-68e2-48f3-888c-d36e7e0c73b0.png">

**Reply by grossmj** [reply]
Hum, I don't see anything wrong. Have you selected "Ubuntu 64-bit Arm" for the operating system?

**Reply by sstah188** [reply]
Hey @grossmj! I'm having the same issue. I selected "Ubuntu 64-bit Arm" for the operating system. 
![image](https://user-images.githubusercontent.com/37384104/145147425-9419994d-c6da-4734-b1a7-a234ef56e7ab.png)

**Reply by sstah188** [reply]
@grossmj @adilnas62711411 I figured out the problem. @adilnas62711411, you have to use the disk images @grossmj provides at the very beginning of the article: 

![image](https://user-images.githubusercontent.com/37384104/145218464-dbc4907c-a6d3-4713-9b44-499dfcb014cb.png)

So I deleted my virtual machine, started over, and downloaded the disk @grossmj provided and now it boots up properly when I start the VM after going through the steps again! Going to go ahead and proceed with the setup, thanks @grossmj!

**Reply by sstah188** [reply]
The problem was that we apparently downloaded the wrong GNS3 VM for this instance. For this experiment this download will not work. 
![image](https://user-images.githubusercontent.com/37384104/145219009-00390d85-e33e-4970-841b-cd491f6eaeac.png)

**Reply by Jverbist** [reply]
Do you have to use an ubuntu arm image as well? or how does it boot?

**Comment by acefer** [comment]
Hi!

I've just followed the guide and tried docker containers with amd64 images. At first, i couldn't make it work properly, but after installing qemu-user-static i was able to run amd64 images on containers.

```sudo apt-get install -y qemu-user-static ```

I've tried with the following images:
This one for routers:
[https://hub.docker.com/r/ajnouri/quagga_alpine/](url)
And this one for computers:
[https://hub.docker.com/r/gns3/endhost](url)

Idk if i misunderstand something or it just works as intended.

**Reply by grossmj** [reply]
Hum, I thought it is not possible to run amd64 images. I will try again, thanks 👍

**Comment by network-shark** [comment]
Wow thank you ! As soon as I get my hands on my m1 , I will help with testing !

**Comment by network-shark** [comment]
I got my Mac and started to compile dynamips from git . Good news it compiled, but without jit . Nevertheless , it seems the idle pc calculation is broken. I got a lot of values and none of them are working. Tested with c7200-advipservicesk9-mz.152-4.S5.bin . @grossmj Does idle pc calculation works for you ?

**Comment by Lifeng-sg** [comment]
Hi,

I am very new to GNS3
I have one Macbook pro ( 2021) running Monterey 12.2, would like to run GNS3 on my Mackbook.
Currently, I have the following apps on it.
1) Parallels Desktop 17 for Mac
2) Ubuntu 20.04.2 ARM64 (installed automatically via Parallels desktop 17)
The Ubuntu is running seem ok.
I am a network guy I would like to know what is next?
1) what is the necessary configuration of Ubuntu
2) is there any procedure for the installation of GNS3? 

thanks 
Lifeng

**Comment by bronsoun** [comment]
@grossmj sincerely I'm sorry but I can't add new device templates due to hash sum mismatch :( what i'd do?

<img width="424" alt="изображение" src="https://user-images.githubusercontent.com/75178164/154763209-361789c3-74d5-40b3-9d59-fa089ac92b2e.png">

**Comment by bejoybkn** [comment]
Hello @grossmj 

Thanks a ton for this post and i finally managed to install GNS3 VM on my M1 (almost lost hope as vmware/VIrtual box support is still officially not there)

Is there any a way i can install [Cisco NXOSv](https://gns3.com/marketplace/appliances/cisco-nx-osv)/[Arista VEOS](https://gns3.com/marketplace/appliances/arista-veos) appliances on this VM as they both are x86?

I tried selecting the different qemu options but its stuck at waiting for the server to respond and does not proceed.

Again massive applauds for releasing this post.

**Comment by network-shark** [comment]
I am using a x86 server to run the vm. On my m1 I start gns3 and connect to the server it runs perfectly. Due to the lack of x86 emulation on m1 I don’t think there will be a solution to run gn3 directly on your mac. Maybe never , until all vendors are shipping vm’s for arm64 and this will take years

**Comment by gfhfhoo** [comment]
Nice solution! @grossmj 
But I still have some problem here:
I want to try to run IOU device on arm platform you mentioned, then, I have installed related libraries of qemu. My question is how can I start my iou device images? Just directly run them by command and will be shown in GNS3 or another way? I don't understand, thank you again. Hope for your reply.

**Comment by cheungtitus** [comment]
>  was able to manually create a template for IOSv-L2 and I had to force to not using KVM by adding -machine accel=tcg otherwise the server would complain that KVM is not available.

@grossmj  Tried this with IOSv using the default QEMU binary which is /usr/bin/qemu-system-x86_64 (v4.2.1) but when the device starts it is stuck in the "Booting from Hard Disk..." stage.
![image](https://user-images.githubusercontent.com/16524925/168441265-570534f7-5cf9-4828-a738-dbd61e427d24.png)

Now if I select to use /usr/bin/qemu-system-aarch64 (v4.2.1) it'd crash with the following in the console.

`QEMU process has stopped, return code: 1`
`Start QEMU with /usr/bin/qemu-system-aarch64 -name IOSv-1 -m 256M -smp cpus=1,sockets=1 -boot order=c -drive file=/opt/gns3/projects/da555b4a-3cef-4778-b7d0-d107722664c9/project-files/qemu/2b3f3a1a-4ad3-44ed-ad82-23357298d9e0/hda_disk.qcow2,if=ide,index=0,media=disk,id=drive0 -uuid 2b3f3a1a-4ad3-44ed-ad82-23357298d9e0 -vnc 0.0.0.0:0 -monitor tcp:127.0.0.1:33681,server,nowait -net none -device e1000,mac=0c:3f:3a:1a:00:00,netdev=gns3-0 -netdev socket,id=gns3-0,udp=127.0.0.1:20001,localaddr=127.0.0.1:20000 -machine accel=tcg`

`Execution log:
qemu-system-aarch64: -machine accel=tcg: No machine specified, and there is no default
Use -machine help to list supported machines`

And with the following pop up.
![image](https://user-images.githubusercontent.com/16524925/168441307-082fdbc7-46ed-4f2e-a59a-d64cabd4859c.png)

Can you comment if there are any additional steps you did?  I'm running this on gns3-vm 2.2.32.

**Comment by parveshtaneja2005** [comment]
getting below error.

GNS3 management console.
Running GNS3 version 2.2.34 on Darwin (64-bit) with Python 3.8.5 Qt 5.15.2 and PyQt 5.15.6.
Copyright (c) 2006-2022 GNS3 Technologies.
Use Help -> GNS3 Doctor to detect common issues.

=> Cannot load the GNS3 VM settings in the preferences dialog: not connected to the controller
Cannot load the IOU license in the preferences dialog: not connected to the controller
Client version 2.2.34 is not the same as server (controller) version 2.2.28
Error while getting templates list: Client version 2.2.34 is not the same as server (controller) version 2.2.28
Error while getting compute list: Client version 2.2.34 is not the same as server (controller) version 2.2.28
Error while getting appliances list: Client version 2.2.34 is not the same as server (controller) version 2.2.28
Error while getting the GNS3 VM settings : Client version 2.2.34 is not the same as server (controller) version 2.2.28
Error while getting the IOU license information: Client version 2.2.34 is not the same as server (controller) version 2.2.28


<img width="1415" alt="Screen Shot 2022-10-04 at 2 01 29 PM" src="https://user-images.githubusercontent.com/114977126/193892504-dd667b7b-6f90-41c6-a43e-9361e6696bc5.png">

**Reply by sangiiii** [reply]
Facing the same issue. Anyone got a workaround for this ?

**Comment by parveshtaneja2005** [comment]
GNS3 VM shows 2.2.28. unable to upgrade it.

gns3@gns3vm:~$ sudo apt upgrade                                              
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
The following package was automatically installed and is no longer required:
  libfwupdplugin1
Use 'sudo apt autoremove' to remove it.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
gns3@gns3vm:~$ 
<img width="1026" alt="Screen Shot 2022-10-04 at 2 02 52 PM" src="https://user-images.githubusercontent.com/114977126/193892900-cfc6b17e-aab1-4dbd-bf78-3a448e98b3f0.png">

**Reply by EagleYing** [reply]
You should run ```sudo apt update``` first.

**Comment by termaulmaul** [comment]
Transport (VMDB) error -14: Pipe connection has been broken

<img width="912" alt="Screenshot 2023-03-08 at 20 34 00" src="https://user-images.githubusercontent.com/109366541/223806527-48473222-4653-4af6-a613-3106fad8356f.png">
<img width="1032" alt="Screenshot 2023-03-08 at 20 37 58" src="https://user-images.githubusercontent.com/109366541/223806551-eb7094ed-a16c-4321-b764-a8c31c1badc4.png">
<img width="752" alt="Screenshot 2023-03-09 at 01 29 04" src="https://user-images.githubusercontent.com/109366541/223806565-08a36dd9-f800-4e7c-a5c1-6f78993d2528.png">
<img width="752" alt="Screenshot 2023-03-09 at 01 29 14" src="https://user-images.githubusercontent.com/109366541/223806568-ca7d181c-a00b-4e02-aed8-c49794c31a3d.png">
<img width="752" alt="Screenshot 2023-03-09 at 01 29 27" src="https://user-images.githubusercontent.com/109366541/223806573-69110ff8-c87d-49c4-993c-a9b99cd31b03.png">
<img width="752" alt="Screenshot 2023-03-09 at 01 30 01" src="https://user-images.githubusercontent.com/109366541/223806574-3483ed49-835b-4de0-abf2-655cd3f6fed8.png">
<img width="1032" alt="Screenshot 2023-03-09 at 01 30 29" src="https://user-images.githubusercontent.com/109366541/223806577-7abd8b85-91fd-4b34-ba7d-349aca068e1d.png">
<img width="692" alt="Screenshot 2023-03-09 at 01 30 41" src="https://user-images.githubusercontent.com/109366541/223806581-7bf9123b-6695-435d-b8db-f50eeaf283ea.png">
<img width="752" alt="image" src="https://user-images.githubusercontent.com/109366541/223806981-2a791ef1-eab2-4b71-962e-db0d834b6d39.png">

**Comment by lhtex** [comment]
Very nice! Thanks!!! Works like a charm!
Now I am waiting for how I can update to 2.2.29 !

**Comment by pmanolak** [comment]
Don't know if I am doing something wrong, but I was never able to start the gns2 vm on fusion in my MacBook M1.
Always failed with the error "This virtual machine cannot be powered on because it requires the X86 machine architecture, which is incompatible with this Arm machine architecture host."
See KB-84273.
I don't think that this is an issue with the Fusion itself but rather than with the GNS3 VM which is compatible only for x86 platforms.
I am using version VMware Fusion 13.0.2 (for Intel-based and Apple silicon Macs).

Any ideas?

**Comment by hanyeolk** [comment]
@grossmj First, Thank you for your all the work. I have followed your steps and was able to install GNS3 VM on my MacBook pro M1 chip. Now, I tried to set up some testing PJ , adding Paloalto VM on GNS3. I found one youtube tutorial about adding PA VM in GNS3 MacBook m1, https://youtu.be/OFu2WXENWdI?si=982vvU1hJD05B912, but when i about to install PA VM, the option "Install the appliance on the GNS3 VM (recommended) is grayed out. Then once i hit next then I get the error message ("The selected server doesn't support KVM.A Linux server of the GNS3 VM running in VMware is required." Please see attached screenshot. Appreciate if anyone can help!!

![Screenshot 2024-09-06 at 11 49 16 AM](https://github.com/user-attachments/assets/4e4d8a89-4b22-431c-9a55-d555776797fc)

![Screenshot 2024-09-06 at 11 49 04 AM](https://github.com/user-attachments/assets/ae4d6061-ccd5-49f8-9643-2a3b96d6c2aa)

**Comment by itcaat** [comment]
After starting IOU I got error IOU VM "IOU1" process has stopped with return code: 255. Anybody got the same error?
<img width="1178" height="650" alt="image" src="https://github.com/user-attachments/assets/fb23197e-aa1b-46e5-9f1a-49cb819182a1" />
